dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+		if ! make_api_request "POST" "$EXIT_API" "Lambda-Extension-Identifier: $EXTENSION_ID" >/dev/null 2>&1; then
+			LOG_WARNING "Failed to notify Lambda platform of extension shutdown"
+		fi
+	fi
+
+	# Clean up temporary files
+	[[ -f $TMP_CHALK_PATH ]] && rm -f "$TMP_CHALK_PATH"
+
+	exit 0
+}
+
+# Setup binary with proper validation and error handling
+setup_chalk_binary() {
+	# Check if chalk binary exists and is executable
+	if [[ ! -x $CHALK_BIN_PATH ]]; then
+		LOG_ERROR "chalk binary not found or not executable at $CHALK_BIN_PATH"
+		return 1
+	fi
+
+	# Copy chalk binary to /tmp if it doesn't exist or is outdated
+	if [[ ! -x $TMP_CHALK_PATH ]] || [[ $CHALK_BIN_PATH -nt $TMP_CHALK_PATH ]]; then
+		LOG_INFO "Copying chalk binary to /tmp"
+		if ! cp "$CHALK_BIN_PATH" "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to copy chalk binary to /tmp"
+			return 1
+		fi
+
+		if ! chmod +x "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to make chalk binary executable in /tmp"
+			return 1
+		fi
+	fi
+
+	return 0
+}
+
+# Process chalk marks with better error handling
+process_chalk_marks() {
+	LOG_INFO "$TMP_CHALK_PATH extracting $CHALK_MARK_PATH"
+	if [[ -f $CHALK_MARK_PATH ]]; then
+		LOG_DEBUG "Found $CHALK_MARK_PATH, running chalk extract."
+		if ! "$TMP_CHALK_PATH" extract "$CHALK_MARK_PATH"; then
+			LOG_WARNING "chalk extract command failed"
+			return 1
+		fi
+	else
+		LOG_WARNING "No chalk.json found at $CHALK_MARK_PATH"
+		return 1
+	fi
+
+	return 0
+}
+
+# JSON parsing using pure bash/standard tools (no jq dependency)
+extract_json_field() {
+	local json="$1"
+	local field="$2"
+
+	# Use grep with improved regex to handle various JSON formatting
+	local result
+	result=$(echo "$json" | grep -o "\"$field\"[[:space:]]*:[[:space:]]*\"[^\"]*\"" |
+		sed "s/\"$field\"[[:space:]]*:[[:space:]]*\"\([^\"]*\)\"/\1/")
+
+	# Handle case where field might have null value
+	if [[ $result == "null" ]]; then
+		result=""
+	fi
+
+	echo "$result"
+}
+
+extract_extension_id() {
+	local temp_file="$1"
+	# Extract the Extension ID from response headers
+	EXTENSION_ID=$(awk '/^[Ll]ambda-[Ee]xtension-[Ii]dentifier:/ {
+        gsub(/\r$/, "");
+        gsub(/^[^:]*:[[:space:]]*/, "");
+        print $0;
+        exit
+    }' "$temp_file")
+
+	rm -f "$temp_file"
+
+	if [[ -z $EXTENSION_ID ]]; then
+		LOG_ERROR "Failed to register extension or parse Extension ID."
+		exit 1
+	fi
+}
+
+register_extension() {
+	# Only register for SHUTDOWN events (INVOKE events are not needed)
+	local registration_payload='{"events":["SHUTDOWN"]}'
+
+	# Make a single registration request with headers included
+	local temp_file
+	temp_file=$(mktemp)
+	local http_code
+
+	http_code=$(curl --silent --write-out "%{http_code}" \
+		--dump-header "$temp_file" \
+		--request POST "$REGISTER_API" \
+		--header "Lambda-Extension-Name: $EXTENSION_NAME" \
+		--header "Content-Type: application/json" \
+		--data "$registration_payload" \
+		--output /dev/null)
+
+	if [[ $http_code -ne 200 ]]; then
```

Comment:
similar error handoling concerns as above.

also as this use `-w` for the output format, if the response does include any useful error message, it wont be showed here as this will only print headers (where there might not be any)

see how setup chalk action does it